zio/csupio/stream.go

```diff
@@ -0,0 +1,31 @@
+package csupio
+
+import (
+	"io"
+	"math"
+	"sync"
+
+	"github.com/brimdata/super/csup"
+)
+
+type stream struct {
+	mu  sync.Mutex
+	r   io.ReaderAt
+	off int64
+}
+
+func (s *stream) next() (*csup.Object, error) {
+	// NewObject puts the right end to the passed in SectionReader and since
+	// the end is unkown just pass MaxInt64.
```

Comment:
1. The meaning of "puts the right end to the passing in SectionReader" isn't clear.
2. s/unkown/unknown
3. Maybe move this below the defer so it's closer to the NewObject call?

Reply:
will update